dart/lib/src/sentry_tracer.dart

```diff
@@ -1,48 +1,69 @@
+import 'dart:async';
+
 import 'package:meta/meta.dart';
 
 import '../sentry.dart';
+import 'sentry_tracer_finish_status.dart';
 
 @internal
 class SentryTracer extends ISentrySpan {
   final Hub _hub;
+  late bool _waitForChildren;
   late String name;
 
-  // missing waitForChildren
-
   late final SentrySpan _rootSpan;
   final List<SentrySpan> _children = [];
   final Map<String, String> _extra = {};
+  Timer? _finishAfterTimer;
+  var _finishStatus = SentryTracerFinishStatus.notFinishing();
 
-  SentryTracer(SentryTransactionContext transactionContext, this._hub) {
+  SentryTracer(SentryTransactionContext transactionContext, this._hub,
+      {bool waitForChildren = false}) {
     _rootSpan = SentrySpan(
       this,
       transactionContext,
       _hub,
       sampled: transactionContext.sampled,
     );
+    _waitForChildren = waitForChildren;
     name = transactionContext.name;
   }
 
   @override
   Future<void> finish({SpanStatus? status}) async {
-    await _rootSpan.finish(status: status);
-
-    // finish unfinished spans otherwise transaction gets dropped
-    for (final span in _children) {
-      if (!span.finished) {
-        await span.finish(status: SpanStatus.deadlineExceeded());
+    _finishAfterTimer?.cancel();
+    _finishStatus = SentryTracerFinishStatus.finishing(status);
+    if (!_rootSpan.finished &&
+        (!_waitForChildren || _haveAllChildrenFinished())) {
+      _rootSpan.status ??= status;
+      await _rootSpan.finish();
+
+      // finish unfinished spans otherwise transaction gets dropped
```

Comment:
This behavior should probably be documented in the method docs.

Reply:
This comment was from before and `SentryTracer` is internal API. Since no other method or field in this class has dedicated documentation, i'm assuming no additional doc is needed here.